projects/storefrontlib/src/cms-structure/outlet/provide-outlet.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  APP_INITIALIZER,
+  ComponentFactoryResolver,
+  FactoryProvider,
+  Type,
+} from '@angular/core';
+import { OutletPosition } from './outlet.model';
+import { OutletService } from './outlet.service';
+
+interface ProvideOutletOptions {
+  name: string;
```

Comment:
I'd change this to `outletRef`

Reply:
Maybe I'm skewed by Angular. To me xxxxRef sounds like ComponentRef, ChangeDetectorRef, ViewRef - reference to the object instance.

To me, this string looks like outlet's ID, name or key.

But I'm not super-oppinionated on it 😉

Reply:
I was thinking about a link to the outletRef directive. But i'm not against just name. `outlet` could also work for me.

Reply:
Agreed offline to `id`